test/python/transpiler/test_light_cone.py

```diff
@@ -0,0 +1,386 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test the LightCone pass"""
+
+import unittest
+from test import QiskitTestCase  # pylint: disable=wrong-import-order
+
+import ddt
+
+from qiskit.circuit import (
+    ClassicalRegister,
+    Parameter,
+    QuantumCircuit,
+    QuantumRegister,
+)
+from qiskit.circuit.library import real_amplitudes
+from qiskit.circuit.library.n_local.efficient_su2 import efficient_su2
+from qiskit.converters import circuit_to_dag
+from qiskit.quantum_info import SparsePauliOp, SparseObservable
+from qiskit.transpiler.passes.optimization.light_cone import LightCone
+from qiskit.transpiler.passmanager import PassManager
+
+
+@ddt.ddt
+class TestLightConePass(QiskitTestCase):
+    """Test the LightCone pass."""
+
+    def __init__(self, methodName: str = "runTest") -> None:
+        super().__init__(methodName)
+
+    @ddt.data("X", "Z")
+    def test_nonparameterized_noncommuting(self, pauli_label):
+        """Test for a non-commuting, asymmetric, weight-one Pauli."""
+        bit_terms = pauli_label
+        light_cone = LightCone(bit_terms=bit_terms, indices=[1])
+        pm = PassManager([light_cone])
+
+        q0 = QuantumRegister(2, "q0")
+        qc = QuantumCircuit(q0)
```

Comment:
You can ofc use registers, but there's not really a reason in these tests. You can also just give the number of qubits.
```suggestion
        qc = QuantumCircuit(2)
```

Reply:
You are right. Now it is fixed.